src/commands/registerCommands.ts

```diff
@@ -181,20 +182,45 @@ async function grammarAssociationCommand(documentURI: Uri, languageClient: Langu
     // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
     const identifier = TextDocumentIdentifier.create(documentURI.toString());
     const grammarURI = fileUri[0];
+    const grammarFilename = path.basename(grammarURI.toString())
+    const currentFilename = path.basename(identifier.uri);
     try {
-      const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-      // Insert the proper syntax for binding
-      const lspTextDocumentEdit = <TextDocumentEdit>result;
-      const workEdits = new WorkspaceEdit();
-      for (const edit of lspTextDocumentEdit.edits) {
-        workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
+      if (bindingType == `fileAssociation`) {
+        // Grammar association through setting.json
+        const grammarAssociationObject = {
```

Comment:
Here it should be nice to show input to fill a pattern file. This field should be pre filled with **/currentFileName

Reply:
Could you clarify what you mean by "pre filled with **/currentFileName"? Are you referring to adding a regex for the file name?

Reply:
I dont remember if file association without wildcard with * s working when your xml file is in a folder of workspace root. If it works forget my comment with wildcard but we need an input field pre filled with file name.

Please.do that and it should be really nice to validate this pattern when user is filling this input field to check it matches the file. To do that we need to provide on lemminx side a new command which have the uri (file name) and pattern   (pattern to validate) as parameter and return a boolean to check it the uri matches the given pattern

Reply:
So consider this case, where the xml file is in a different directory from the schema file.

For example, if your directory is as follows:
-dir1
>-file.xsd
>-dir2
>>-file.xml

and you select file.xsd as the associated file for file.xml, the file association object would currently be:
`{
"pattern": "file.xml",
"systemId": file.xsd"
}`

Do you want instead that the user selects the relative path for 'file.xml'? i.e. replaces the ** in **/file.xml?

Reply:
In this case I would like to have an input field pre-filled with **/file.xml and that's it.

Reply:
If the user doesn't change the **, should the default be the same directory?